apps/admin/frontend/src/components/import_err_file_modal.tsx

```diff
@@ -57,62 +45,59 @@ export function ImportElectionsResultReportingFileModal({
 
   assert(electionDefinition);
   assert(isElectionManagerAuth(auth) || isSystemAdministratorAuth(auth));
-  const [currentState, setCurrentState] = useState<ModalState>({
-    state: 'init',
-  });
 
-  async function handleImportElectionResultReportingFile(path: string) {
+  function handleImportElectionResultReportingFile(path: string) {
     const filepath = path;
-    setCurrentState({ state: 'loading' });
-
-    const result = await importElectionResultReportingFileMutation.mutateAsync({
+    importElectionResultReportingFileMutation.mutate({
       precinctId,
       ballotStyleId,
       votingMethod,
       filepath,
     });
-
-    if (result.isOk()) {
-      setCurrentState({ state: 'success' });
-    } else {
-      setCurrentState({ state: 'error', filename: filepath });
-    }
   }
 
-  const processElectionResultReportingFileFromFilePicker: InputEventFunction =
-    async (event) => {
-      // electron adds a path field to the File object
-      assert(window.kiosk, 'No window.kiosk');
-      const input = event.currentTarget;
-      const files = Array.from(input.files || []);
-      const file = files[0] as ElectronFile;
-
-      if (!file) {
-        onClose();
-        return;
-      }
+  const processElectionResultReportingFileFromFilePicker: InputEventFunction = (
+    event
+  ) => {
+    // electron adds a path field to the File object
+    assert(window.kiosk, 'No window.kiosk');
+    const input = event.currentTarget;
+    const files = Array.from(input.files || []);
+    const file = files[0] as ElectronFile;
+
+    if (!file) {
+      onClose();
+      return;
+    }
 
-      await handleImportElectionResultReportingFile(file.path);
-    };
+    handleImportElectionResultReportingFile(file.path);
+  };
 
-  if (currentState.state === 'error') {
+  function errorContents(message: string) {
     return (
       <Modal
-        title="Error"
-        content={
-          <P>
-            There was an error reading the contents of{' '}
-            <Font weight="bold">{currentState.filename}</Font>:{' '}
-            {currentState.errorMessage}
-          </P>
-        }
+        title="Failed to Import Results"
+        content={<P>{message}</P>}
         onOverlayClick={onClose}
         actions={<Button onPress={onClose}>Close</Button>}
       />
     );
   }
 
-  if (currentState.state === 'success') {
+  /* istanbul ignore next */
+  if (importElectionResultReportingFileMutation.isError) {
```

Comment:
Will this error just get caught by the error boundary due to the react-query default settings?

Reply:
Good point - [a6cc1c0](https://github.com/votingworks/vxsuite/pull/5344/commits/a6cc1c0da44cf8db044956ea99b016705c37d632)